src/Data/Container/Indexed/Examples/MultiSortedAlgebra.agda

```diff
@@ -0,0 +1,433 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Multi-sorted algebras as indexed containers
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --without-K --guardedness #-}  -- forced by Agda-2.6.2-rc1
+
+module Data.Container.Indexed.Examples.MultiSortedAlgebra where
+
+-- Preliminaries
+-- =============
+
+-- We import library content for indexed containers, standard types, and setoids.
+
+open import Level
+
+open import Data.Container.Indexed.Core            using (Container; ⟦_⟧; _◃_/_)
+open import Data.Container.Indexed.FreeMonad       using (_⋆C_)
+open import Data.W.Indexed                         using (W; sup)
+
+open import Data.Product                           using (Σ; _×_; _,_; Σ-syntax); open Σ
+open import Data.Sum                               using (_⊎_; inj₁; inj₂; [_,_])
+open import Data.Empty.Polymorphic                 using (⊥; ⊥-elim)
+
+open import Function                               using (_∘_)
+open import Function.Bundles                       using (Func)
+
+open import Relation.Binary                        using (Setoid; IsEquivalence)
+open import Relation.Binary.PropositionalEquality  using (_≡_; refl)
+
+import Data.Container.Indexed.Relation.Binary.Equality.Setoid as ICSetoid
+import Relation.Binary.Reasoning.Setoid as SetoidReasoning
+
+open Setoid using (Carrier; _≈_; isEquivalence)
+open Func renaming (f to apply)
+
+-- Letter ℓ denotes universe levels.
+
+variable
+  ℓ ℓ' ℓˢ ℓᵒ ℓᵃ ℓᵐ ℓᵉ ℓⁱ : Level
+  I : Set ℓⁱ
+  S : Set ℓˢ
+
+-- The interpretation of a container (Op ◃ Ar / sort) is
+--
+--   ⟦ Op ◃ Ar / sort ⟧ X s = Σ[ o ∈ Op s ] ((i : Ar o) → X (sort o i))
+--
+-- which contains pairs consisting of an operator $o$ and its collection
+-- of arguments.  The least fixed point of (X ↦ ⟦ C ⟧ X) is the indexed
+-- W-type given by C, and it contains closed first-order terms of the
+-- multi-sorted algebra C.
+
+-- We need to interpret indexed containers on Setoids.
+-- This definition is missing from the standard library v1.7.
+
+⟦_⟧s : (C : Container I S ℓᵒ ℓᵃ) (ξ : I → Setoid ℓᵐ ℓᵉ) → S → Setoid _ _
+⟦ C ⟧s ξ = ICSetoid.setoid ξ C
+
+-- Multi-sorted algebras
+-- =====================
```

Comment:
Bikeshedding, but we try to keep a consistent comment style in the library. Double header:
```agda
------------------------------------------------------------------------
-- Multi-sorted algebras
------------------------------------------------------------------------
```
and single headers:
```agda
------------------------------------------------------------------------
-- Model
```